receiver/awscontainerinsightreceiver/internal/host/machineinfo.go

```diff
@@ -25,65 +25,131 @@ import (
 type MachineInfo struct {
 	sync.RWMutex
 	logger          *zap.Logger
+	nodeCapacity    *NodeCapacity
+	ec2metadata     *Ec2metadata
+	ebsVolume       *EbsVolume
+	ec2Tags         *Ec2Tags
 	refreshInterval time.Duration
 	shutdownC       chan bool
 }
 
 // NewMachineInfo creates a new MachineInfo struct
 func NewMachineInfo(refreshInterval time.Duration, logger *zap.Logger) *MachineInfo {
+	nodeCapacity, err := NewNodeCapacity(logger)
+	if err != nil {
+		logger.Error("Failed to initialize NodeCapacity", zap.Error(err))
+	}
+
 	mInfo := &MachineInfo{
+		nodeCapacity:    nodeCapacity,
+		ec2metadata:     NewEc2metadata(refreshInterval, logger),
 		refreshInterval: refreshInterval,
 		shutdownC:       make(chan bool),
 		logger:          logger,
 	}
 
-	// TODO: add more initializations
+	mInfo.lazyInitEbsVolume()
+	mInfo.lazyInitEc2Tags()
 	return mInfo
 }
 
+func (m *MachineInfo) lazyInitEbsVolume() {
+	if m.ebsVolume == nil {
+		//delay the initialization. If instance id is not available, ebsVolume is set to nil
+		//Because ebs volumes only change occasionally, we refresh every 5 collection intervals to reduce ec2 api calls
+		m.ebsVolume = NewEbsVolume(m.GetInstanceID(), 5*m.refreshInterval, m.logger)
+	}
+
+	go func() {
+		refreshTicker := time.NewTicker(m.refreshInterval)
+		defer refreshTicker.Stop()
+		for {
+			select {
+			case <-refreshTicker.C:
+				if m.ebsVolume != nil {
+					return
+				}
+				m.logger.Info("refresh to initialize ebsVolume")
+				m.ebsVolume = NewEbsVolume(m.GetInstanceID(), m.refreshInterval, m.logger)
+			case <-m.shutdownC:
+				return
+			}
+		}
+	}()
+}
```

Comment:
This doesn't feel right.  If initializing the EBS volume detector depends on the EC2 instance ID then initializing the volume detector shouldn't be attempted until the instance ID is in hand.  Perhaps have the EC2 metadata detector put the instance ID on a channel when it finds it since that appears to be its only responsibility.